txstratum/healthcheck.py

```diff
@@ -0,0 +1,265 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from hathorlib.client import HathorClient
+
+if TYPE_CHECKING:
+    from txstratum.manager import TxMiningManager
+
+
+class ComponentType(str, Enum):
+    """Enum used to store the component types that can be used in the HealthCheckComponentStatus class."""
+
+    DATASTORE = "datastore"
+    INTERNAL = "internal"
+    FULLNODE = "fullnode"
+
+
+class HealthCheckStatus(str, Enum):
+    """Enum used to store the component status that can be used in the HealthCheckComponentStatus class."""
+
+    PASS = "pass"
+    WARN = "warn"
+    FAIL = "fail"
+
+
+@dataclass
+class ComponentHealthCheck:
+    """This class is used to store the result of a health check in a specific component."""
+
+    component_name: str
+    component_type: ComponentType
+    status: HealthCheckStatus
+    output: str
+    time: Optional[str] = None
+    component_id: Optional[str] = None
+    observed_value: Optional[str] = None
+    observed_unit: Optional[str] = None
+
+    def update(self, new_values: Dict[str, Any]) -> None:
+        """
+        Update the object with the new values passed as kwargs.
+
+        Also updates the time field with the current time with the format YYYY-MM-DDTHH:mm:ssZ
+        """
+        self.time = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        for key, value in new_values.items():
+            setattr(self, key, value)
```

Comment:
Do you see any advantage?

Reply:
Well, I was hoping `replace()` could help with type safety, but I tested it and it doesn't. In this case, it would only simplify the code a bit, as now it could simply be `replace(self, **new_values)` instead of the loop. But then you would have to return the new instance instead of updating the current one in place, which I do think it's better (i always prefer immutability), but would require a small refactor in you code and I don't see it as necessary.